### PR TITLE
add replication id

### PR DESCRIPTION
**Pull request by oriyarde**

### CODE REVIEW

File: go.mod [hunk 1]
```diff
@@ -16,3 +16,5 @@ require (
 	k8s.io/client-go v0.22.1
 	sigs.k8s.io/controller-runtime v0.9.6
 )
+
+replace github.com/csi-addons/spec => github.com/oriyarde/spec v0.1.1-0.20210919134406-35c5f35c5e90
```
Comment (Madhu-1) [inline go.mod:20]: https://github.com/csi-addons/spec/releases/tag/v0.1.1 release is out. make use of it.

File: controllers/volumereplication_controller.go [hunk 2]
```diff
@@ -358,12 +363,15 @@ func (r *VolumeReplicationReconciler) waitForVolumeReplicationResource(logger lo
 }
 
 // markVolumeAsPrimary defines and runs a set of tasks required to mark a volume as primary
-func (r *VolumeReplicationReconciler) markVolumeAsPrimary(volumeReplicationObject *replicationv1alpha1.VolumeReplication, logger logr.Logger, volumeID string, parameters, secrets map[string]string) error {
+func (r *VolumeReplicationReconciler) markVolumeAsPrimary(volumeReplicationObject *replicationv1alpha1.VolumeReplication,
+	logger logr.Logger, volumeID string, replicationId string, parameters, secrets map[string]string) error {
```
Comment (Madhu-1) [inline controllers/volumereplication_controller.go:367]: replicationId to `replicationID` keeping ID caps

File: pkg/client/fake/replication-client.go [hunk 3]
```diff
@@ -23,38 +23,38 @@ import (
 // ReplicationClient to fake replication operations.
 type ReplicationClient struct {
 	// EnableVolumeReplicationMock mocks EnableVolumeReplication RPC call.
-	EnableVolumeReplicationMock func(volumeID string, secrets, parameters map[string]string) (*replicationlib.EnableVolumeReplicationResponse, error)
+	EnableVolumeReplicationMock func(volumeID string, replicationId string, secrets, parameters map[string]string) (*replicationlib.EnableVolumeReplicationResponse, error)
```
Comment (Madhu-1) [inline pkg/client/fake/replication-client.go:26]: ```volumeID , replicationID string```  use ID instead of `Id`

**Review by Madhu-1 — COMMENTED** [review]
changes LGTM. @oriyarde can you please squash the commits? mergify bot is not configured to squash and Merge the PR's

File: controllers/volumereplication_controller.go (hunk 2 above)
Comment (oriyarde) [reply controllers/volumereplication_controller.go]: done

File: pkg/client/fake/replication-client.go (hunk 3 above)
Comment (oriyarde) [reply pkg/client/fake/replication-client.go]: done

File: go.mod (hunk 1 above)
Comment (oriyarde) [reply go.mod]: done

**Comment by oriyarde** [comment]
> changes LGTM. @oriyarde can you please squash the commits? mergify bot is not configured to squash and Merge the PR's

let's continue in https://github.com/csi-addons/volume-replication-operator/pull/118 instead

**Pull request closed without merging**